### PR TITLE
Modularity. Alternate names.

### PR DESCRIPTION
**Pull request by JustusAdam**
Added the option for alternate names and made code more modular.

### CODE REVIEW

File: scripts/mensa.coffee [hunk 1]
```diff
@@ -16,26 +16,87 @@
...
     mensa = msg.match[1]
-    index = mensen.indexOf(mensa.toLowerCase()) + 79
-    if index == 78
-      msg.send "Kenne leider keine solche Mensa..."
-      return
-    getMeals(mensa, index, robot, msg)
+    generic_resp_func(mensa, (m) -> msg.send m)
 
   robot.respond /mensen/i, (msg) ->
     msg.send "Ich kann dir heutige Speisepläne für die folgenden Mensen holen:\n - #{mensen.join('\n - ')}\nSprich' mich einfach mit `matthias mensa <mensa>` an."
```
Comment (kiliankoe) [inline scripts/mensa.coffee]: This just lists a whole bunch of `[object Object]`, you're going to want to make this list the new names of the objects.

Comment (JustusAdam) [inline scripts/mensa.coffee]: Geez, I totally forgot to test that function

**Comment by kiliankoe** [comment]
I'm also getting a traceback for the mensa zittau, not that anyone needs that, but any clue why?

```
matthias> matthias mensa zittau
matthias> [Thu Jul 30 2015 20:42:00 GMT+0200 (CEST)] ERROR SyntaxError: Unexpected end of input
  at Object.parse (native)
  at /Users/kilian/Desktop/matthias/scripts/mensa.coffee:86:9, <js>:73:21
  at IncomingMessage.<anonymous> (/Users/kilian/Desktop/matthias/node_modules/hubot/node_modules/scoped-http-client/src/index.js:95:22)
  at IncomingMessage.emit (events.js:129:20)
  at _stream_readable.js:908:16
  at process._tickCallback (node.js:355:11)
```

This is the output from openmensa:

```
[
  {
    "id": 1432339,
    "name": "Schweinesteak mit Rahmchampignons, dazu Buttererbsen und Petersilienkartoffeln",
    "category": "Angebote",
    "prices": {
      "students": 2.45,
      "employees": 4.14,
      "pupils": null,
      "others": null
    },
    "notes": [
      "Menü enthält Knoblauch",
      "Menü enthält Schweinefleisch"
    ]
  },
  {
    "id": 1432340,
    "name": "Vegetarisches Schnitzel Toscana mit einer fruchtigen Tomatensoße und Butterspätzle",
    "category": "Angebote",
    "prices": {
      "students": 2.3,
      "employees": 3.99,
      "pupils": null,
      "others": null
    },
    "notes": [
      "Menü enthält kein Fleisch"
    ]
  }
]
```

**Comment by JustusAdam** [comment]
It is actually empty. If you command out line 87 & 88 and instead just print the body with the callback function it prints nothing.
The body is absolutely empty.

**Comment by JustusAdam** [comment]
Wooooow ... what a giant rabbit hole am I digging up here right now ...
- Mensa zittau is actually nr [89](http://openmensa.org/c/89)
- Mensa U-Boot is actually rn [88](http://openmensa.org/c/88)

and so on

Until "Zelt" is actully not 79, but rather 78 ... could it been that you made an error there? Because I just copied from your offset (which was 79). Did you forget that arrays start with index 0?

90 is actually Zittaus second mensa, which is currently out of order

I'll fix the code to resolve all that this will probably also resolve the issue of "wrong data" :smile:

**Comment by kiliankoe** [comment]
Well... Oops^^ I just copy-pasted stuff from the openmensa API output and might've fucked that up^^

**Comment by JustusAdam** [comment]
Actually, your default mensa used to be 79, which is in fact the alte mensa, so that was correct. But the offset on the array lookup was 79 also which is wrong, because 78 is zelt. I am fixing it right now.
And I am also future proofing the system against empty responses. To make sure it doesn't just crash

**Comment by JustusAdam** [comment]
I've added myself to the authors, if that's ok with you.

**Comment by JustusAdam** [comment]
Oh. and it should be ready to be merged now. ... I think.

**Comment by kiliankoe** [comment]
Sorry, didn't keep up, movie night and stuff ;)

Looks good and works nice, merging it. Thanks!

**Pull request merged**